indra/newview/llselectmgr.cpp

```diff
@@ -1996,9 +1996,55 @@ bool LLSelectMgr::selectionSetGLTFMaterial(const LLUUID& mat_id)
                     asset_id = BLANK_MATERIAL_ASSET_ID;
                 }
             }
+
+            // Preserve existing texture entry transforms when switching to PBR material
+            LLTextureEntry* tep = objectp->getTE(te);
+            F32 existing_scale_s = 1.0f, existing_scale_t = 1.0f;
+            F32 existing_offset_s = 0.0f, existing_offset_t = 0.0f;
+            F32 existing_rotation = 0.0f;
+            bool should_preserve_transforms = false;
+            LLGLTFMaterial* preserved_override = nullptr;
+
+            if (tep && asset_id.notNull())
+            {
+                // Check if we have non-default texture entry transforms to preserve
+                tep->getScale(&existing_scale_s, &existing_scale_t);
+                tep->getOffset(&existing_offset_s, &existing_offset_t);
+                existing_rotation = tep->getRotation();
```

Comment:
This looks wrong. I don't think material offsets are stored there, they should be in override material.